e2e_playwright/st_graphviz_chart_test.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction, wait_for_app_run
+
+
+def test_initial_setup(app: Page):
+    """Initial setup: ensure charts are loaded."""
+
+    title_count = len(app.locator(".stGraphVizChart > svg > g > title").all())
+    assert title_count == 5
+
+
+def test_shows_left_and_right_graph(app: Page):
+    """Test if it shows left and right graph."""
+
+    left_text = app.locator(".stGraphVizChart > svg > g > title").nth(3).text_content()
+    right_text = app.locator(".stGraphVizChart > svg > g > title").nth(4).text_content()
+    assert "Left" in left_text and "Right" in right_text
+
+
+def test_renders_with_specified_engines(
+    app: Page, assert_snapshot: ImageCompareFunction
```

Comment:
nit: Do we apply any style changes to graphviz charts when in dark mode? If so, it might be worth it to use `themed_app` here instead to automatically do the snapshot tests in light and dark mode.

Reply:
We don't make any style changes for graphviz charts based on the theme afaik